### DISCUSSION TITLE
Deployment: new zk-forge version creates errors when deploying

### DISCUSSION THREAD

**Discussion by L0GYKAL** [zkSync-Community-Hub/zksync-developers#770] [Deployments] [answered]
### Team or Project

Zyfi

### Environment

Mainnet

### zkSolc Version

0.8.21

### zksync-ethers Version

N/A

### Minimal Reproducible Code or Repo Link

I already created an issue on the zk-forge repo:
https://github.com/matter-labs/foundry-zksync/issues/720
And here is the example I created to show the issue: https://github.com/matter-labs/foundry-zksync/issues/720#issuecomment-2468142169

### Does this work on other EVMs? (If yes, please list at least 1 of them)

Arbitrum

### Description of What Your Contract Does

Upgradable ERC20 using UUPS

**Comment by alhenaai** [comment]
It seems you are encountering issues with deploying an upgradable ERC20 contract using UUPS on the ZKsync mainnet with the new zk-forge version. Since you've already created an issue on the zk-forge repository, the best course of action is to follow up there for specific guidance related to the error you're experiencing.

For further assistance and to discuss this issue with other developers, I recommend visiting the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions). This platform can provide more in-depth answers and community support for your deployment challenges.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by popzxc** [comment]
Thanks for reporting! The maintainers of foundry-zksync are looking into this issue.

**Comment by dutterbutter** [comment]
Will be resolved with https://github.com/matter-labs/foundry-zksync/pull/727

**Reply by MexicanAce** [reply, marked as answer]
The [latest nightly release](https://github.com/matter-labs/foundry-zksync/releases/tag/nightly-dc26b62957d8f07032d7b2be80284d9197c2e13a) of `foundry-zksync` should have this fix. Please let us know if you run into any further issues by marking this Discussion as `unanswered` 👍
